hazelcast/proxy/multi_map.py

```diff
@@ -445,6 +447,52 @@ def put(self, key: KeyType, value: ValueType) -> Future[bool]:
         request = multi_map_put_codec.encode_request(self.name, key_data, value_data, thread_id())
         return self._invoke_on_key(request, key_data, multi_map_put_codec.decode_response)
 
+    def put_all(self, map: typing.Dict[KeyType, typing.Sequence[ValueType]]) -> Future[bool]:
+        """Stores given values by associating with corresponding keys in dictionary format.
+
+        Warning:
+            This method uses ``__hash__`` and ``__eq__`` methods of binary form
+            of the key, not the actual implementations of ``__hash__`` and
+            ``__eq__`` defined in key's class.
+
+        Args:
+            map: the map corresponds to multimap entries.
+
+        Returns:
+            ``True`` if size of the multimap is increased, ``False`` if the
+            multimap already contains the key-value tuple.
+        """
+        check_not_none(map, "map can't be None")
+        if not map:
+            return ImmediateFuture(None)
+
+        partition_service = self._context.partition_service
+        partition_map: typing.Dict[int, typing.List[typing.Tuple[Data, typing.List[Data]]]] = {}
+
+        for key, values in map.items():
+            try:
+                check_not_none(key, "key can't be None")
+                check_not_none(values, "values can't be None")
+                serialized_key = self._to_data(key)
+                serialized_values = []
+                for value in values:
+                    check_not_none(value, "value can't be None")
+                    serialized_values.append(self._to_data(value))
+                partition_id = partition_service.get_partition_id(serialized_key)
+                try:
+                    partition_map[partition_id].append((serialized_key, serialized_values))
+                except KeyError:
+                    partition_map[partition_id] = [(self._to_data(key), serialized_values)]
+            except SchemaNotReplicatedError as e:
+                return self._send_schema_and_retry(e, self.put_all, map)
+
+        futures = []
+        for partition_id, entry_list in partition_map.items():
+            request = multi_map_put_all_codec.encode_request(self.name, entry_list)
+            future = self._invoke_on_partition(request, partition_id)
+            futures.append(future)
+        return combine_futures(futures)
```

Comment:
combine futures returns a future that resolves to a list. the method signature and the return type does not match here

Reply:
Changed the return view [87a6da9](https://github.com/hazelcast/hazelcast-python-client/pull/573/commits/87a6da958a9cfc1bc57fff31af7995cf7081020e)

Reply:
but you are still returning something not None. `return combine_futures(futures)` that returns a future that will resolve to a list once it is done. you need to add some sort of continuation with `continue_with` method to make it return none